stl/inc/numeric

```diff
@@ -70,19 +70,30 @@ inline constexpr bool _Plus_on_arithmetic_ranges_reduction_v = false;
 #endif // _STD_VECTORIZE_WITH_FLOAT_CONTROL
 
 template <class _InIt, class _Ty, class _BinOp>
-_NODISCARD _Ty reduce(const _InIt _First, const _InIt _Last, _Ty _Val, _BinOp _Reduce_op) {
+_NODISCARD
+#if defined(_HAS_CXX20) && defined(__cpp_lib_is_constant_evaluated)
+constexpr
+#endif
+_Ty reduce(const _InIt _First, const _InIt _Last, _Ty _Val, _BinOp _Reduce_op) {
     // return commutative and associative reduction of _Val and [_First, _Last), using _Reduce_op
     _Adl_verify_range(_First, _Last);
     auto _UFirst      = _Get_unwrapped(_First);
     const auto _ULast = _Get_unwrapped(_Last);
+#if defined(_HAS_CXX20) && defined(__cpp_lib_is_constant_evaluated)
+    if (_STD is_constant_evaluated()) {
+        for (; _UFirst != _ULast; ++_UFirst) {
+            _Val = _Reduce_op(_STD move(_Val), *_UFirst);
+        }
+        return _Val;
+    }
+#endif
     if constexpr (_Plus_on_arithmetic_ranges_reduction_v<_Unwrapped_t<const _InIt&>, _Ty, _BinOp>) {
         (void) _Reduce_op; // TRANSITION, VSO-486357

```

Comment:
I tried the same earlier the week with adding the `is_constant_evaluated` branch into the constexpr and removing the else. However,  I am unsure which version I like best
```cpp
if constexpr (_Plus_on_arithmetic_ranges_reduction_v<_Unwrapped_t<const _InIt&>, _Ty, _BinOp>) {
#ifdef __cpp_lib_is_constant_evaluated
        if (!_STD is_constant_evaluated())
#endif // __cpp_lib_is_constant_evaluated
        {
            (void) _Reduce_op; // TRANSITION, VSO-486357
            return _Reduce_plus_arithmetic_ranges(_UFirst, _ULast, _Val);
        }
    }
    for (; _UFirst != _ULast; ++_UFirst) {
        _Val = _Reduce_op(_STD move(_Val), *_UFirst); // Requirement missing from N4713
    }

    return _Val;
```